frontend/src/components/managePermissions/UserDataTable.vue

```diff
@@ -0,0 +1,315 @@
+<script setup lang="ts">
+import { reactive, ref } from 'vue';
+import IconCapitol from '@/components/common/IconCapitol.vue';
+import { FilterMatchMode } from 'primevue/api';
+import Column from 'primevue/column';
+import DataTable from 'primevue/datatable';
+import InputText from 'primevue/inputtext';
+import { IconSize } from '@/enum/IconEnum';
+import { useConfirm } from 'primevue/useconfirm';
+import ConfirmDialog from 'primevue/confirmdialog';
+import type { PropType } from 'vue';
+import type { FamApplicationUserRoleAssignmentGet } from 'fam-api/dist/model/fam-application-user-role-assignment-get';
+
+const confirm = useConfirm();
+
+const props = defineProps({
+    loading: {
+        type: Boolean,
+        default: false,
+    },
+    userRoleAssignments: {
+        type: [Array] as PropType<
+            FamApplicationUserRoleAssignmentGet[] | undefined
+        >,
+        required: true,
+    },
+    selectedApplicationDisplayText: {
+        type: String,
+        requried: true,
+    },
+    isApplicationSelected: {
+        type: Boolean,
+        required: true,
+        default: false,
+    },
+});
+
+const filters = ref({
+    global: { value: null, matchMode: FilterMatchMode.CONTAINS },
+    'user.user_name': { value: null, matchMode: FilterMatchMode.STARTS_WITH },
+    'role.parent_role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.client_number.forest_client_number': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+});
+
+const confirmDeleteData = reactive({
+    userName: '',
+    role: '',
+});
+
+const emit = defineEmits<{
+    (
+        e: 'deleteUserRoleAssignment',
+        item: FamApplicationUserRoleAssignmentGet
+    ): void;
+}>();
+
+function deleteAssignment(assignment: FamApplicationUserRoleAssignmentGet) {
+    confirmDeleteData.role = assignment.role.role_name;
+    confirmDeleteData.userName = assignment.user.user_name;
+    confirm.require({
+        group: 'deleteAssignment',
+        header: 'Remove Access',
+        rejectLabel: 'Cancel',
+        acceptLabel: 'Remove',
+        acceptClass: 'p-button-danger',
+        accept: () => {
+            emit('deleteUserRoleAssignment', assignment);
+        },
+    });
+}
+</script>
+
+<template>
+    <ConfirmDialog group="deleteAssignment">
+        <template #message>
+            <p>
+                Are you sure you want to remove
+                <strong>{{ confirmDeleteData.role }}</strong> access to
+                <strong>{{ confirmDeleteData.userName }}</strong> in
+                <strong>{{ selectedApplicationDisplayText }}</strong>
+            </p>
+        </template>
+    </ConfirmDialog>
+    <div class="data-table-container">
+        <span class="custom-data-table-bg-layer"></span>
```

Comment:
Hi Jota,
It looks strange to me this line.
I am curious what this <span> does when it is just by itself?
(`<span class="custom-data-table-bg-layer"></span>`)

Reply:
For the table to align with the body automatic was easier to make the background layer a separate component.